project/src/components/room-page/room-page.jsx

```diff
@@ -1,22 +1,38 @@
-import React from 'react';
-import { useHistory } from 'react-router-dom';
-import { useLocation } from 'react-router-dom';
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useHistory, useParams } from 'react-router-dom';
 import { connect } from 'react-redux';
-import { AppRoute } from '../../const.js';
-import { getRatingInPercents, mapOffersToClient } from '../../utils.js';
+import { AppRoute, AuthorizationStatus } from '../../const.js';
+import { getRatingInPercents } from '../../utils.js';
 import offerType from '../offers-prop/offers-prop.js';
-import reviewsType from '../reviews-props/reviews-props.js';
 import Header from '../header/header.jsx';
 import FormComment from '../form-comment/form-comment.jsx';
 import ReviewsList from '../reviews/reviews-list.jsx';
 import MapRoomPage from '../map/room-page-map.jsx';
 import OtherPlaces from '../other-places/other-places-list.jsx';
+import NotFoundPage from '../not-found-page/not-found-page.jsx';
+import { fetchNearbyList, fetchComments } from '../../store/api-actions.js';
 function RoomPage(props) {
-  const { offers, reviews } = props;
+  const { offers, authorizationStatus, getId } = props;
   const history = useHistory();
-  const location = useLocation();
+  const { id } = useParams();
+
+  const getOfferLinkFoundPage = () => {
+    const offerLink = offers.map((it) => Number(it.id)).includes(Number(id));
```

Comment:
```suggestion
    const hasOffer = offers.some((it) => Number(it.id) === Number(id));
```
Метод some тут подходит лучше
И это так же лучше вынести в какую-то функцию хелпер. Чтобы сделать компонент чище (на проверку есть ли оофер с таким иди или нет)